wgpu-types/src/lib.rs

```diff
@@ -934,5 +934,52 @@ pub enum TextureComponentType {
     Uint,
 }
 
+impl From<TextureFormat> for TextureComponentType {
+    fn from(format: TextureFormat) -> Self {
+        match format {
+            TextureFormat::R8Unorm
+            | TextureFormat::R8Uint
+            | TextureFormat::R16Uint
+            | TextureFormat::Rg8Unorm
```

Comment:
normalized formats (both unorm and snorm) should map to `Float`

Reply:
Ahhh I see! I've added a commit that makes this change.